Fix off-by-one in get_road()

rikardfalkeborn opened the pull request:
When the input is int8_t, getRoad() expects it to go from 1 to 6 (inclusive).

rikardfalkeborn committed:
Fix off-by-one when getting road

When the input is int8_t, getRoad expects it to go from 1 to 6 (inclusive).

Noordfrees reviewed (approved):
LGTM, not tested. Good catch :)

rikardfalkeborn committed:
Use enum values for directions

rikardfalkeborn committed:
Don't hardcode number of elements in road

Since it's a c-array with known size, use range-based for loops,
std::fill() and std::any_of() to avoid opencoding the size of the
array.

gunchleoc committed:
Merge branch 'master' into fix-off-by-one-in-get_road

gunchleoc committed:
Fix codecheck

gunchleoc commented:
We are getting test suite failures:

`widelands: ../src/economy/flag.cc:847: virtual void Widelands::Flag::cleanup(Widelands::EditorGameBase&): Assertion '!road' failed.`

Instructions on how to run the test suite at https://www.widelands.org/wiki/RegressionTests/

rikardfalkeborn committed:
Don't use range for loop when removing elements

rikardfalkeborn commented:
> We are getting test suite failures:

I'm pretty sure I tested, but only a release build...

gunchleoc commented:
Your last commit fixed it, so this can go in once the builds are finished :)
